autoscoper/src/net/Socket.cpp

```diff
@@ -272,11 +281,11 @@ void Socket::deleteConnection()
 {
   //std::cerr << "client disconnected" << std::endl;
   QTcpSocket * obj = dynamic_cast<QTcpSocket *>(sender());
-  if (obj)
+  if (obj && obj != NULL)
```

Comment:
```suggestion
  if (obj != NULL)
```

In the future, we will modernize the codebase through clang-tidy. In the meantime, using `NULL` (instead of `nullptr`) makes sense and is consistent with the the remaining of the codebase.

Reply:
Considering moving the fix to `Socket::deleteConnection` in a dedicated commit labeled `BUG: ...`, then the update of the API will be done in a following up commit starting with `ENH: ...`

Reply:
The issue that this resolves was introduced because of the changes I made in the previous commit, it was not happening beforehand.